Adds WebCodecs VideoFrame (#8722)
mdn/content: files/en-us/web/api/videoframe/clone/index.md

---
title: VideoFrame.clone()
slug: Web/API/VideoFrame/clone
tags:
  - API
  - Method
  - Reference
  - clone
  - VideoFrame
browser-compat: api.VideoFrame.clone
---
{{DefaultAPISidebar("Web Codecs API")}}

> The **`clone`** method of the {{domxref("VideoFrame")}} interface creates a new `VideoFrame` object with reference to the same media resource as the original.

Author: Yes that's how the spec describes it. See: https://www.w3.org/TR/webcodecs/#media-resource

I think it's clear that the actual pixel data is one thing and that these interfaces reference it.

> The **`clone`** method of the {{domxref("VideoFrame")}} interface creates a new `VideoFrame` object with reference to the same media resource as the original.

Reviewer: ```suggestion
The **`clone`** method of the {{domxref("VideoFrame")}} interface creates a new `VideoFrame` object that references the same media resource as the original.
```
